ambry-account/src/main/java/com/github/ambry/account/InMemoryUnknownAccountService.java

```diff
@@ -26,6 +27,7 @@
 class InMemoryUnknownAccountService implements AccountService {
   private static final Collection<Account> accounts =
       Collections.unmodifiableCollection(Collections.singletonList(Account.UNKNOWN_ACCOUNT));
+  private final CopyOnWriteArraySet<AccountUpdateListener> accountUpdateListeners = new CopyOnWriteArraySet<>();
```

Comment:
First a concurrent version of `Set` is needed, and you are correct that this implementation is optimized for reading, as it is expected adding/removing from the `set` would not be as frequent as reading.